assets/src/edit-story/components/panels/textStyle/panel.js

```diff
@@ -0,0 +1,237 @@
+  const textDecoration = getCommonValue(selectedElements, 'textDecoration');
+  const bold = getCommonValue(selectedElements, 'bold');
+
+  const padding = getCommonValue(selectedElements, 'padding') ?? '';
+
+  // Font settings.
+  const fontFamily = getCommonValue(selectedElements, 'fontFamily');
+  const fontSize = getCommonValue(selectedElements, 'fontSize');
+  const fontWeight = getCommonValue(selectedElements, 'fontWeight');
+  const fontWeights = getCommonValue(selectedElements, 'fontWeights');
+  const fontFallback = getCommonValue(selectedElements, 'fontFallback');
+
+  // Color settings.
+  const color = getCommonValue(selectedElements, 'color');
+  const backgroundColor = getCommonValue(selectedElements, 'backgroundColor');
+  const backgroundOpacity = getCommonValue(
+    selectedElements,
+    'backgroundOpacity'
+  );
+  const textOpacity = getCommonValue(selectedElements, 'textOpacity');
+
+  const {
+    actions: { getFontWeight, getFontFallback },
+  } = useFont();
+
+  const [state, setState] = useState({
+    backgroundColor,
+    backgroundOpacity,
+    bold,
+    color,
+    fontFamily,
+    fontStyle,
+    fontSize,
+    fontWeight,
+    fontFallback,
+    fontWeights,
+    textDecoration,
+    textAlign,
+    textOpacity,
+    letterSpacing,
+    lineHeight,
+    padding,
+  });
+  const [lockPaddingRatio, setLockPaddingRatio] = useState(true);
+  useEffect(() => {
+    const currentFontWeights = getFontWeight(fontFamily);
+    const currentFontFallback = getFontFallback(fontFamily);
+    setState({
+      backgroundColor,
+      backgroundOpacity,
+      bold,
+      color,
+      textOpacity,
+      textAlign,
+      letterSpacing,
+      lineHeight,
+      padding,
+      fontFamily,
+      fontStyle,
+      fontSize,
+      fontWeight,
+      fontWeights: currentFontWeights,
+      fontFallback: currentFontFallback,
+      textDecoration,
+    });
+  }, [
+    bold,
+    color,
+    textOpacity,
+    textAlign,
+    letterSpacing,
+    lineHeight,
+    padding,
+    getFontWeight,
+    fontFamily,
+    getFontFallback,
+    fontStyle,
+    fontSize,
+    fontWeight,
+    textDecoration,
+    backgroundColor,
+    backgroundOpacity,
+  ]);
+  const handleSubmit = (evt) => {
+    onSetProperties(state);
+    onSetProperties((properties) => {
+      const {
+        padding: oldPadding,
+        width,
+        height: oldHeight,
+        rotationAngle,
+        x,
+        y,
+      } = properties;
+      const { padding: newPadding } = state;
+      const updatedState = removeUnsetValues(state);
+      const newProperties = { ...properties, ...updatedState };
+      const newHeight = dataPixels(calculateTextHeight(newProperties, width));
+      const [dx, dy] = calcRotatedResizeOffset(
+        rotationAngle,
+        0,
+        0,
+        0,
+        newHeight - oldHeight
+      );
+      const ratio = getPaddingRatio(oldPadding.horizontal, oldPadding.vertical);
```

Comment:
When I am playing around in the panel and changing some settings, the editor constantly crashes because `oldPadding` is `undefined`

Reply:
Yep, this PR needs quite a bit of updating after various merges, also migration, etc.. Marked it WIP for now, will re-request review once ready.